contracts/DxMgnPool.sol

```diff
@@ -76,25 +74,26 @@ contract DxMgnPool is Ownable {
 
     function withdrawDeposit() public returns(uint) {
         require(currentState == State.DepositWithdrawnFromDx || currentState == State.MgnUnlocked, "Funds not yet withdrawn from dx");
-        
+        require(!hasParticpationWithdrawn[msg.sender],"sender has already withdrawn funds");
+
         uint totalDepositAmount = 0;
         Participation[] storage participations = participationsByAddress[msg.sender];
         uint length = participations.length;
         for (uint i = 0; i < length; i++) {
             totalDepositAmount += calculateClaimableDeposit(participations[i]);
-            participations[i].withdrawn = true;
         }
+        hasParticpationWithdrawn[msg.sender] = true;
         SafeERC20.safeTransfer(address(depositToken), msg.sender, totalDepositAmount);
         return totalDepositAmount;
     }
 
     function withdrawMagnolia() public returns(uint) {
         require(currentState == State.MgnUnlocked, "MGN has not been unlocked, yet");
-
+        require(hasParticpationWithdrawn[msg.sender], "Withdraw deposits first");
+        
         uint totalMgnClaimed = 0;
         Participation[] memory participations = participationsByAddress[msg.sender];
         for (uint i = 0; i < participations.length; i++) {
-            require(participations[i].withdrawn, "Withdraw deposits first");
             totalMgnClaimed += calculateClaimableMgn(participations[i]);
         }
         delete participationsByAddress[msg.sender];
```

Comment:
Should we `delete hasParticpationWithdrawn[msg.sender]` here as well?

Reply:
Yes, I will open another PR